pkg/controller/operands/operand.go

```diff
@@ -145,6 +146,14 @@ func (h *genericOperand) addCrToTheRelatedObjectList(req *common.HcoRequest, fou
 		return err
 	}
 
+	existingRef, err := objectreferencesv1.FindObjectReference(req.Instance.Status.RelatedObjects, *objectRef)
+	if err != nil {
+		return err
+	}
+	if existingRef != nil && !reflect.DeepEqual(existingRef, *objectRef) {
```

Comment:
@tiraboschi @nunnatsa Actually, we can set StatusDirty=true when there is no existingRef as well to be in the safe side. It is not necessary at the moment since it is covered by another logic in another part.